### ISSUE TITLE
Schema Audit

### ISSUE THREAD

**Issue by armsnyder** [gitlabhq/terraform-provider-gitlab#553]
I noticed some inconsistencies in our schemas
* Inconsistencies between attribute names and/or structures between the provider schemas and the upstream GitLab API schemas. (Example: [gitlab_user](https://github.com/gitlabhq/terraform-provider-gitlab/blob/18e9201a44dc2f7867eaf503654b5d923b9b6df7/gitlab/resource_gitlab_user.go#L42) attribute `is_admin` vs [API](https://docs.gitlab.com/ce/api/users.html#user-creation) field `admin`)
* Inconsistencies between attribute names across different resources in this provider, for attributes that have the same function. (Example: `project` vs `id` vs `project_id` vs `full_path` all being used to mean a project name/id)

Correcting inconsistencies will have these benefits
* More consistency between this provider and other API tools users may be using to manage GitLab.
* Increasing the number of resources that could be updated using code generation as the GitLab API evolves.

This will have to be a breaking change and a new major release. I'm thinking of coordinating this with upgrading to the [SDK v2](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html).

**Comment by timofurrer** [comment]
I think it doesn't make sense to do this now with SDK v2, since users are actually not affected by the SDK upgrade ... Let's schedule a proper v4 release and assign things there we want to have in :)

**Comment by armsnyder** [comment]
Yep my opinion on that changed given how long sdk v2 has been supported in terraform core 👍

**Comment by armsnyder** [comment]
`group` vs `group_id` is another inconsistency 😭

**Comment by Shocktrooper** [comment]
@armsnyder Should we start making a tabled list in the description of this issue? 1 column being what variations there are of a specific variable/attribute, another being what files/resources it exists in and the third column being what the agreed upon final value should be?
